google/oauth2/_client.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""OAuth 2.0 client.
+
+This is a client for interacting with an OAuth 2.0 authorization server's
+token endpoint.
+
+For more information about the token endpoint, see
+`Section 3.1 of rfc6749`_
+
+.. _Section 3.1 of rfc6749: https://tools.ietf.org/html/rfc6749#section-3.2
+"""
+
+import datetime
+import json
+
+from six.moves import http_client
+from six.moves import urllib
+
+from google.auth import _helpers
+from google.auth import exceptions
+
+_JWT_GRANT_TYPE = 'urn:ietf:params:oauth:grant-type:jwt-bearer'
+_REFRESH_GRANT_TYPE = 'refresh_token'
+
+
+def _handle_error_response(response_body):
+    """"Translates an error response into an exception.
+
+    Args:
+        response_body (str): The decoded response data.
+
+    Raises:
+        google.auth.exceptions.RefreshError
+    """
+    try:
+        error_data = json.loads(response_body)
+        error_details = ': '.join([
+            error_data['error'],
+            error_data.get('error_description')])
+    # If not details could be extracted, use the response data.
+    except (KeyError, ValueError):
```

Comment:
`ValueError` is for `json.loads` and `KeyError` is for `error_data['error']`?

I'm unclear if error payloads look like

```json
{
  "error": "...",
  ...
}
```

or

```json
{
  "error": {
    "errors": [
      ...
    ],
    ...
  }
  ...
}
```


e.g. https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2093#issuecomment-241438784 vs. e.g. a 403 with `https://www.googleapis.com/storage/v1/b/haha`

```json  
{
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "accountDisabled",
    "message": "The account for bucket \"haha\" has been disabled.",
    "locationType": "header",
    "location": "Authorization"
   }
  ],
  "code": 403,
  "message": "The account for bucket \"haha\" has been disabled."
 }
}
```